diff --git a/filterPWM.py b/filterPWM.py
index f4aea24..ae82eba 100644
--- a/filterPWM.py
+++ b/filterPWM.py
@@ -3,6 +3,7 @@ from multiprocessing import Pool
 import numpy as np
 import logging
 import sys
+import shutil
 
 from utils import calculate_H_model_bg, calculate_H_model, model_sim, update_models, filter_pwms, parse_meme, write_meme
 
@@ -43,6 +44,12 @@ def main():
 
     # parse input query meme file
     model_set = parse_meme(query_file)
+
+    if len(model_set['models']) < 2:
+        # no real filtering possible, copy input to output
+        shutil.copyfile(query_file, out_file)
+        return
+
     # pre-compute entropy for all the models
     models = update_models(model_set['models'])
 
